client/client.go

```diff
@@ -23,14 +46,82 @@ func NewFurtiveClient(ws *websocket.Conn) *FurtiveClient {
 
 func (fc *FurtiveClient) ReadMessages() {
 	for {
-		var msg *Message
+		var contents json.RawMessage
+		msg := &Message{
+			Contents: &contents,
+		}
 		if err := fc.ws.ReadJSON(&msg); err != nil {
 			fc.ws.Close()
 			log.Fatalln("Error when reading JSON:", err)
 			return
 		}
-		fmt.Println("New message:", msg.Contents)
-		fc.SendMessage(&Message{fmt.Sprintf("READMSG:%s", msg.Contents)})
+
+		switch msg.Type {
+		case "votingData":
+			var votingData *VotingData
+			if err := json.Unmarshal(contents, &votingData); err != nil {
+				log.Fatalln("Error when reading JSON:", err)
+				return
+			}
+			fc.RoundOne(votingData)
+		case "roundOne":
+			var values *Values
+			if err := json.Unmarshal(contents, &values); err != nil {
+				log.Fatalln("Error when reading JSON:", err)
+				return
+			}
+			fc.AfterRoundOne(values)
+			fc.RoundTwo()
+		case "roundTwo":
+			var values *Values
+			if err := json.Unmarshal(contents, &values); err != nil {
+				log.Fatalln("Error when reading JSON:", err)
+				return
+			}
+			fc.CheckResult(values)
+		default:
+			log.Fatalf("unknown message type: %q", msg.Type)
+		}
+	}
+}
+
+func (fc *FurtiveClient) RoundOne(votingData *VotingData) {
+	fmt.Println("Question:", votingData.Question)
+	participant := NewParticipant(votingData.Generator, votingData.BigPrimary, votingData.Divisor, votingData.Id)
+	gXi := participant.BroadcastGXi()
+	value := &Value{gXi}
+	msg := &Message{"roundOne", value}
+	fc.SendMessage(msg)
+	fc.Participant = participant
+}
+
+func (fc *FurtiveClient) AfterRoundOne(values *Values) {
+	fc.Participant.ComputeGYi(values.Numbers, values.Length)
+}
+
+func (fc *FurtiveClient) RoundTwo() {
+	reader := bufio.NewReader(os.Stdin)
+	fmt.Print("Do you want to veto? (Y/N): ")
+	text, _ := reader.ReadString('\n')
+	var vote *big.Int
```

Comment:
Create `value := &Value{}` here and assign to `Number` property in if statement

Reply:
yeah, that's neater